backend/product-listing.go

```diff
@@ -67,13 +72,52 @@ type JsonRoot struct {
 }
 
 var products []Product
+var shoppingCarts map[string]ShoppingCart
+var clientIds *list.List
 
 func InitProductListing() {
 	initProducts(DIST_FOLDER + "/json/related_products.json")
 	RegisterSample(SHOPPING_CART, renderShoppingCart)
 	RegisterSample("samples_templates/product_listing", renderProductListing)
 	RegisterSample("samples_templates/product", renderProduct)
 	RegisterSampleEndpoint("samples_templates/product_listing", SEARCH, handleSearchRequest)
+	http.HandleFunc(ADD_TO_CART_PATH, func(w http.ResponseWriter, r *http.Request) {
+		handlePost(w, r, addToCart)
+	})
+	shoppingCarts = make(map[string]ShoppingCart)
+	clientIds = list.New()
+}
+
+func addToCart(w http.ResponseWriter, r *http.Request) {
+	EnableCors(w, r)
+	response := ""
+	name := r.FormValue("name")
+	quantity := r.FormValue("quantity")
+	clientId := r.FormValue("clientId")
+	img := r.FormValue("img")
+	price := r.FormValue("price")
+
+	shoppingCartItem := ShoppingCartItem{
+		Img:      img,
+		Name:     name,
+		Price:    price,
+		Quantity: quantity,
+	}
+
+	if clientIds.Len() == 100 {
+		delete(shoppingCarts, clientIds.Front().Value.(string))
```

Comment:
You could use an existing LRU Cache implementation, e.g.: https://github.com/golang/groupcache/blob/master/lru/lru.go